### PR TITLE
Add MVN fusion

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Add different patterns that can be fused to MVN-6*

### Tickets:
 - *24552*
 - *28819*

### CODE REVIEW

**Comment by mvafin** [comment]
CI failures look not connected with the changes.

**Review by vgavrilo — APPROVED** [review]

**Review by sadolini — APPROVED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+    // Sqrt(ReduceMean((x - ReduceMean(x, axes)) ^ 2))
+    // `--Power--------------------------------------'
+    auto const_0_5 = pattern::wrap_type<ngraph::opset6::Constant>();
+    auto power_sqrt = pattern::wrap_type<opset6::Power>({ mean3, const_0_5 }, single_consumer);
+
+    auto sqrt = pattern::wrap_type<opset6::Sqrt>({ mean3 }, single_consumer);
+
+    const auto powerOrSqrt = std::make_shared<pattern::op::Or>(OutputVector{ power_sqrt, sqrt });
+
+    // (Sqrt(ReduceMean((x - ReduceMean(x, axes)) ^ 2)) + eps)
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp:71]: If the difference in this transformations only in `sqrt(...) + eps` and `sqrt(... + eps)` so I would suggest to untie this patterns using `Or`.

File: inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+                                                pattern_to_output.at(power).get_node_shared_ptr(),
+                                                pattern_to_output.at(mean3).get_node_shared_ptr(),
+                                                pattern_to_output.at(add_eps).get_node_shared_ptr() });
+
+        if (pattern_to_output.count(mean2) != 0 && pattern_to_output.count(sub2) != 0) {
+            nodes_to_copy_info.push_back(pattern_to_output.at(mean2).get_node_shared_ptr());
+            nodes_to_copy_info.push_back(pattern_to_output.at(sub2).get_node_shared_ptr());
+        }
+
+        if (pattern_to_output.count(cast) != 0) {
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp:314]: And in all other places.
```suggestion
        if (pattern_to_output.count(cast)) {
```

**Review by ilyachur — APPROVED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp (hunk 1 above)
Comment (mvafin) [reply inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp]: I thought about that, but it is pretty hard to read those patterns, so I thought that splitting is fine

**Review by pavel-esir — APPROVED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        if (axes_1_value != axes_3_value) {
+            return false;
+        }
+        if (pattern_to_output.count(mean2_axes)) {
+            auto axes_2_node = std::dynamic_pointer_cast<ngraph::opset6::Constant>(pattern_to_output.at(mean2_axes).get_node_shared_ptr());
+            if (!axes_2_node) {
+                return false;
+            }
+            auto axes_2_value = axes_2_node->cast_vector<int64_t>();
+            if (!(axes_1_value == axes_2_value)) {
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp:134]: nitpick
```suggestion
            if (axes_1_value != axes_2_value) {
```

File: inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            }
+        } else if (pattern_to_output.count(powerOrSqrt_is)) {
+            mode = op::MVNEpsMode::INSIDE_SQRT;
+            nodes_to_copy_info.push_back(pattern_to_output.at(add_eps_is).get_node_shared_ptr());
+            if (pattern_to_output.count(power_sqrt_is)) {
+                nodes_to_copy_info.push_back(pattern_to_output.at(power_sqrt_is).get_node_shared_ptr());
+            } else if (pattern_to_output.count(sqrt_is)) {
+                nodes_to_copy_info.push_back(pattern_to_output.at(sqrt_is).get_node_shared_ptr());
+            }
+        }
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp:161]: I think we need to add `else` to return false.

File: inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+            nodes_to_copy_info.push_back(pattern_to_output.at(sub2).get_node_shared_ptr());
+        }
+
+        if (pattern_to_output.count(cast)) {
+            nodes_to_copy_info.push_back(pattern_to_output.at(cast).get_node_shared_ptr());
+        }
+
+        if (pattern_to_output.count(div_alt)) {
+            nodes_to_copy_info.push_back(pattern_to_output.at(div_alt).get_node_shared_ptr());
+        } else if (pattern_to_output.count(power_div) != 0 && pattern_to_output.count(div) != 0) {
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/mvn_fusion.cpp:175]: nitpick
```suggestion
        } else if (pattern_to_output.count(power_div) && pattern_to_output.count(div)) {
```

**Review by GlebKazantaev — APPROVED** [review]
In general ok, just some minor comments.

**Comment by ilyachur** [comment]
Please fix CI

**Comment by mvafin** [comment]
Had to force rebase, because couldn't revert submodule CMake changes.

**Pull request merged**